src/CryptState.cpp

```diff
@@ -237,6 +240,18 @@ void CryptState::ocb_encrypt(const unsigned char *plain, unsigned char *encrypte
 		AESencrypt(tmp, tmp, &encrypt_key);
 		XOR(reinterpret_cast<subblock *>(encrypted), delta, tmp);
 		XOR(checksum, checksum, reinterpret_cast<const subblock *>(plain));
+
+		// Counter-cryptanalysis described in section 9 of https://eprint.iacr.org/2019/311
+		// For an attack, the second to last block (i.e. the last iteration of this loop)
+		// must be all 0 except for the last byte (which may be 0 - 128).
+		if (len - AES_BLOCK_SIZE <= AES_BLOCK_SIZE) {
+			unsigned char sum = 0;
+			for (int i=0;i<AES_BLOCK_SIZE-1;i++) {
```

Comment:
```suggestion
			for (auto i = 0; i < AES_BLOCK_SIZE-1; ++i) {
```

Reply:
why would you use `auto` in this case? This doesn't make much sense to me and it's actually longer to type than `int`

Reply:
I would actually prefer `size_t`, `auto` expands to `int` in this case.

Reply:
The index should be of same type as `AES_BLOCK_SIZE` (whatever that is) imo